pkg/registry/batch/job/strategy.go

```diff
@@ -236,8 +243,9 @@ func generateSelector(obj *batch.Job) {
 	if obj.Spec.Selector.MatchLabels == nil {
 		obj.Spec.Selector.MatchLabels = make(map[string]string)
 	}
-	if _, found := obj.Spec.Selector.MatchLabels["controller-uid"]; !found {
-		obj.Spec.Selector.MatchLabels["controller-uid"] = string(obj.ObjectMeta.UID)
+
+	if _, found := obj.Spec.Selector.MatchLabels[batch.ControllerUid]; !found {
+		obj.Spec.Selector.MatchLabels[batch.ControllerUid] = string(obj.ObjectMeta.UID)
```

Comment:
So the name was never actually used in the selector?

I wonder if the name label is relic from a time when we use to match on job name instead of UID.

Reply:
https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/batch/job/strategy.go#L211

Maybe.  This is the only comment I can find that explains the rational.